src/main/java/nl/knaw/dans/dvcli/command/DatasetRoleAssignment.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (C) 2024 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.dvcli.command;
+
+import nl.knaw.dans.dvcli.action.ConsoleReport;
+import nl.knaw.dans.dvcli.action.Pair;
+import nl.knaw.dans.dvcli.action.ThrowingFunction;
+import nl.knaw.dans.lib.dataverse.DatasetApi;
+import nl.knaw.dans.lib.dataverse.DataverseException;
+import nl.knaw.dans.lib.dataverse.model.RoleAssignment;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+import picocli.CommandLine.ParentCommand;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Command(name = "role-assignment",
+         mixinStandardHelpOptions = true,
+         description = "Manage role assignments on one or more datasets.")
+public class DatasetRoleAssignment extends AbstractCmd {
+    @ParentCommand
+    private DatasetCmd datasetCmd;
+
+    static class AllArgs {
+        @ArgGroup(exclusive = false)
+        AssignmentCommand assignmentCommand;
+
+        @Option(names = "list", description = "List role assignments for specified dataset")
+        boolean list;
+    }
+
+    @ArgGroup(multiplicity = "1")
+    private AllArgs allArgs;
+
+    static class AssignmentCommand {
+        static class AssignmentAction {
+            @Option(names = "add", description = "add role assignment to specified dataset(s)'")
```

Comment:
This hack is unnecessary. The picocli library supports subcommands. `add`, `remove` and `list` are not options but subcommands and should be implemented that way.

Reply:
The Role Assignment commands are separated in sub-commands now.

Reply:
@janvanmansum If you are satisfied with the changes, you could close this conversation and accept the PR.